galley/pkg/crd/validation/config.go

```diff
@@ -27,55 +27,107 @@ import (
 
 	"github.com/ghodss/yaml"
 	"k8s.io/api/admissionregistration/v1beta1"
+	"k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	admissionregistration "k8s.io/client-go/kubernetes/typed/admissionregistration/v1beta1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
 
 	"istio.io/istio/pkg/log"
 )
 
-// Check if the galley validation endpoint is ready to receive requests.
-func (wh *Webhook) endpointReady() error {
-	endpoints, err := wh.clientset.CoreV1().Endpoints(wh.deploymentNamespace).Get(wh.deploymentName, metav1.GetOptions{})
-	if err != nil {
-		return fmt.Errorf("%s/%v endpoint ready check failed: %v", wh.deploymentNamespace, wh.deploymentName, err)
+func endpointReady(store cache.Store, queue workqueue.RateLimitingInterface, namespace, name string) (shutdown, ready bool) {
+	key, quit := queue.Get()
```

Comment:
Nit: Consider changing the order of the return variables. That aligns better with (value, error) type of idiomatic returns in Go.